src/System.Linq/tests/EmptyEnumerable.cs

```diff
@@ -61,5 +61,14 @@ public void CastToArray()
             var emptyEnumerable = Enumerable.Empty<object>();
             Assert.Same(emptyEnumerable, (object[])emptyEnumerable);
         }
+
+        [Fact]
+        public void CachedEnumerator()
+        {
+            var emptyEnumerable1 = Enumerable.Empty<object>();
+            var emptyEnumerable2 = (IList<object>)emptyEnumerable1;
+
+            Assert.Same(emptyEnumerable1.GetEnumerator(), emptyEnumerable2.GetEnumerator());
+        }
```

Comment:
Instead of moving this test here, how about deleting it and instead adding the line:
```
Assert.Same(enumerable1.GetEnumerator(), enumerable2.GetEnumerator());
```
to the end of ```TestEmptyCached<T>()```.  That way we verify this for all of the tests that use that helper with various Ts.